control/grpc.py

```diff
@@ -0,0 +1,445 @@
+                context.set_details(f"{ex}")
+            return pb2.req_status()
+
+        if context:
+            # Update gateway state
+            try:
+                json_req = json_format.MessageToJson(
+                    request, preserving_proto_field_name=True)
+                self.gateway_state.add_bdev(bdev_name, json_req)
+            except Exception as ex:
+                self.logger.error(
+                    f"Error persisting create_bdev {bdev_name}: {ex}")
+                raise
+
+        return pb2.req_status(status=True)
+
+    def delete_bdev(self, request, context=None):
+        """Deletes a bdev."""
+
+        self.logger.info(
+            f"Received request to delete bdev {request.bdev_name}")
+        try:
+            ret = self.spdk_rpc.bdev.bdev_rbd_delete(
+                self.spdk_rpc_client,
+                request.bdev_name,
+            )
+            self.logger.info(f"delete_bdev {request.bdev_name}: {ret}")
+        except Exception as ex:
+            self.logger.error(f"delete_bdev failed with: \n {ex}")
+            if context:
+                context.set_code(grpc.StatusCode.INTERNAL)
+                context.set_details(f"{ex}")
+            return pb2.req_status()
+
+        if context:
+            # Update gateway state
+            try:
+                self.gateway_state.remove_bdev(request.bdev_name)
+            except Exception as ex:
+                self.logger.error(
+                    f"Error persisting delete_bdev {request.bdev_name}: {ex}")
+                raise
+
+        return pb2.req_status(status=ret)
+
+    def create_subsystem(self, request, context=None):
+        """Creates a subsystem."""
+
+        self.logger.info(
+            f"Received request to create subsystem {request.subsystem_nqn}")
+        try:
+            ret = self.spdk_rpc.nvmf.nvmf_create_subsystem(
+                self.spdk_rpc_client,
+                nqn=request.subsystem_nqn,
+                serial_number=request.serial_number,
+            )
+            self.logger.info(f"create_subsystem {request.subsystem_nqn}: {ret}")
+        except Exception as ex:
+            self.logger.error(f"create_subsystem failed with: \n {ex}")
+            if context:
+                context.set_code(grpc.StatusCode.INTERNAL)
+                context.set_details(f"{ex}")
+            return pb2.req_status()
+
+        if context:
+            # Update gateway state
+            try:
+                json_req = json_format.MessageToJson(
+                    request, preserving_proto_field_name=True)
+                self.gateway_state.add_subsystem(request.subsystem_nqn,
+                                                 json_req)
+            except Exception as ex:
+                self.logger.error(f"Error persisting create_subsystem" +
+                                  f" {request.subsystem_nqn}: {ex}")
+                raise
+
+        return pb2.req_status(status=ret)
```

Comment:
`pb2.subsystem_info` would pass back the request parameter for subsystem nqn (as opposed to one returned from the SPDK RPC call), so I thought it was especially not needed.

Reply:
Does `nvmf_create_subsystem` SPDK RPC return subsystem NQN?  https://spdk.io/doc/jsonrpc.html gives the following "example response":
```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": true
}
```

Reply:
No, it returns status of the RPC call.

Reply:
What did you mean by "as opposed to one returned from the SPDK RPC call" then?

Reply:
I meant that if the SPDK RPC call returned a subsystem nqn, this would be more useful to return back from the gRPC than the request nqn. However, as with `rbd_bdev_create` and the returned bdev name, it would still not be needed. Therefore a case where the request nqn is returned is especially not needed. Sorry for being unclear earlier.